osf/models/files.py

```diff
@@ -445,7 +445,7 @@ def update(self, revision, data, user=None):
                 break
         else:
             # Insert into history if there is no matching etag
-            if data.get('modified'):
+            if data.get('modified') and None not in [item['modified'] for item in self.history]:
```

Comment:
You can wrap the error in a `dev_mode` if statement to ensure it doesn't go to production. But yeah, probably logging to sentry with a fairly distinctive title and putting in testing notes to check sentry would work well.